### ISSUE TITLE
Direnv environments are not picked up with either fish or nushell used

### ISSUE THREAD

**Issue by WeetHet** [zed-industries/zed#8633]
### Current status
- Fish fix is merged
- Nushell fix is yet to be found
- Should we also ship fixes for stuff like elvish/oil/etc?

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The current way to pick up the environment as used in 
https://github.com/zed-industries/zed/blob/dab886f4794e8d14cc82dc8e4906e93b78aa8fe8/crates/project/src/project.rs#L9366-L9385

does not work with neither fish nor nu shells. This is because direnv has some weird behaviour preventing it from evaluating. The dirty fix would be to force evaluation:
```rust
let additional_command = match PathBuf::from_str(shell.as_str()) {
    Ok(shell_path) if shell_path.file_name() == Some(OsStr::new("fish")) => {
        Some("eval (direnv export fish)")
    }
    Ok(shell_path) if shell_path.file_name() == Some(OsStr::new("nu")) => {
        Some("direnv export json | from json | default {} | load-env")
    }
    _ => None,
};

let command = match additional_command {
    Some(additional_command) => format!(
        "cd {dir:?}; {}; echo {marker}; /usr/bin/env -0; exit 0;",
        additional_command
    ),
    None => format!("cd {dir:?}; echo {marker}; /usr/bin/env -0; exit 0;"),
};
```
or something like that. 

### Environment

Zed: v0.125.1 (Zed Preview)
OS: macOS 14.3.1
Memory: 8 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

<img width="977" alt="Screenshot 2024-02-29 at 22 00 12" src="https://github.com/zed-industries/zed/assets/43210583/8e37d4bf-6d42-4331-a36e-e6ab97f6a185">

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by WeetHet** [comment]
@mrnugget, this seems like your stuff

**Comment by WeetHet** [comment]
@mrnugget I have discovered a horrible hack for fish: add `emit fish_prompt` after cd, which triggers direnv

**Comment by WeetHet** [comment]
Or do `set -x __direnv_export_again; emit fish_preexec;`
<img width="914" alt="Screenshot 2024-03-01 at 10 15 35" src="https://github.com/zed-industries/zed/assets/43210583/6c68432f-a13c-486b-91ca-e61bf73310de">

**Comment by WeetHet** [comment]
> Just to confirm this: you're running into this when trying to use `gopls`, right? Because no other place uses this method yet.

Yeah, but its a more global problem with some tools (like direnv) requiring `fish_prompt` event to activate

**Comment by mrnugget** [comment]
Got it. I think special-casing `fish` and using `emit fish_prompt` after `cd` isn't that hacky. Why do you think it's hacky? What am I missing?

**Comment by WeetHet** [comment]
> Got it. I think special-casing `fish` and using `emit fish_prompt` after `cd` isn't that hacky. Why do you think it's hacky? What am I missing?

We need to do that for every shell, I still don't know what should be done for nushell

**Comment by WeetHet** [comment]
We can do fish for now and leave this open for suggestions on nushell

**Comment by JosephTLyons** [comment]
This is available in Zed [v0.126.0-pre](https://zed.dev/releases/preview/0.126.0).

**Comment by WeetHet** [comment]
@JosephTLyons please reopen as this is still not implemented for nushell. Also labeling this as help needed would be great as I have no idea how to fix nushell
